Add support for Java >= 9

ennerf opened the pull request:
This pull request adds compatibility for Jigsaw and fixes #16. Tested on JDK 9 / 10 / 11.

* added a default Jigsaw `module-name`

* replaced `sun.misc.Cleaner` workaround with [Glue's](https://github.com/HebiRobotics/Glue) multi-release jar

ennerf commented:
I'm not familiar with Travis. Is there a way to access `file:///home/travis/build/diffplug/matfilerw/build/findbugs/main.html`  to see what rule violations it found?

nedtwigg commented:
If you run `gradlew check` on your machine, you can look at the `build/findbugs/main.html` on your own machine

ennerf commented:
Thanks. It was complaining about the manual `System.gc()` which has been in there for a while. Not sure how that got through the checks before.

I also ran into issues with the ProGuard-Maven plugin and MR-Jars, so I removed the dependency again and added a reflective version.

nedtwigg commented:
LGTM.  Ready for me to merge and release?

ennerf commented:
Go for it

nedtwigg commented:
Releases as 3.1.0
